packages/client-sdk-nodejs/src/config/vector-configuration.ts

```diff
@@ -0,0 +1,98 @@
+import {Configuration} from './configuration';
+import {MomentoLoggerFactory} from '@gomomento/sdk-core';
+import {TransportStrategy} from './transport';
+import {Middleware} from './middleware/middleware';
+import {RetryStrategy} from './retry/retry-strategy';
+
+export interface VectorConfigurationProps {
+  /**
+   * Configures logging verbosity and format
+   */
+  loggerFactory: MomentoLoggerFactory;
+  /**
+   * Configures how and when failed requests will be retried
+   */
+  retryStrategy: RetryStrategy;
+  /**
+   * Configures low-level options for network interactions with the Momento service
+   */
+  transportStrategy: TransportStrategy;
+  /**
+   * Configures middleware functions that will wrap each request
+   */
+  middlewares: Middleware[];
+}
+
+export class VectorConfiguration implements Configuration {
+  private readonly loggerFactory: MomentoLoggerFactory;
+  private readonly retryStrategy: RetryStrategy;
```

Comment:
I left this out of the Python client since we weren't using it.

Reply:
yup, removing in subsequent PR, didn't catch it during this one